packages/shopping-cart/src/shopping-cart-actions.ts

```diff
@@ -0,0 +1,36 @@
+import { createRequestCartProducts } from './create-request-cart-product';
+import type {
+	RequestCartProduct,
+	DispatchAndWaitForValid,
+	ShoppingCartManagerActions,
+} from './types';
+
+export function createActions( dispatch: DispatchAndWaitForValid ): ShoppingCartManagerActions {
+	return {
+		reloadFromServer: () => dispatch( { type: 'CART_RELOAD' } ),
+		removeCoupon: () => dispatch( { type: 'REMOVE_COUPON' } ),
+		addProductsToCart: async ( products ) =>
```

Comment:
This function and `replaceProductsInCart` doesn't need to be `async` as it doesn't do any `await` and it returns a promise anyway.

Reply:
It actually does serve a purpose since these functions call another function that can throw an Error. I actually just discovered this issue recently thanks to some of the tests I added as part of this PR. See https://github.com/Automattic/wp-calypso/pull/55564 which explains it in more detail.

Reply:
I see, the `createRequestCartProduct` function can throw an exception and we need to convert that to a rejected promise. :+1: